packages/grid/x-data-grid-premium/src/tests/cellSelection.DataGridPremium.test.tsx

```diff
@@ -0,0 +1,305 @@
+import * as React from 'react';
+import { spy } from 'sinon';
+import { expect } from 'chai';
+import { getCell } from 'test/utils/helperFn';
+// @ts-ignore Remove once the test utils are typed
+import { createRenderer, fireEvent, act } from '@mui/monorepo/test/utils';
+import {
+  DataGridPremium,
+  DataGridPremiumProps,
+  GridApi,
+  useGridApiRef,
+  gridClasses,
+} from '@mui/x-data-grid-premium';
+import { getBasicGridData } from '@mui/x-data-grid-generator';
+
+describe('<DataGridPremium /> - Cell Selection', () => {
+  const { render } = createRenderer();
+
+  let apiRef: React.MutableRefObject<GridApi>;
+
+  function TestDataGridSelection({
+    rowLength = 4,
+    ...other
+  }: Omit<DataGridPremiumProps, 'rows' | 'columns' | 'apiRef'> &
+    Partial<Pick<DataGridPremiumProps, 'rows' | 'columns'>> & { rowLength?: number }) {
+    apiRef = useGridApiRef();
+
+    const data = React.useMemo(() => getBasicGridData(rowLength, 3), [rowLength]);
+
+    return (
+      <div style={{ width: 300, height: 300 }}>
+        <DataGridPremium
+          {...data}
+          {...other}
+          apiRef={apiRef}
+          rowSelection={false}
+          unstable_cellSelection
+          disableVirtualization
+        />
+      </div>
+    );
+  }
+
+  it('should select the cell clicked', () => {
+    render(<TestDataGridSelection />);
+    expect(document.querySelector('.Mui-selected')).to.equal(null);
+    const cell = getCell(0, 1);
+    fireEvent.click(cell);
+    expect(document.querySelector('.Mui-selected')).to.equal(cell);
+  });
+
+  it('should unselect already selected cells when selecting a cell', () => {
+    render(<TestDataGridSelection />);
+    const cell01 = getCell(0, 1);
+    fireEvent.click(cell01);
+    expect(cell01).to.have.class('Mui-selected');
+    const cell11 = getCell(1, 1);
+    fireEvent.click(cell11);
+    expect(cell01).not.to.have.class('Mui-selected');
+    expect(cell11).to.have.class('Mui-selected');
+  });
+
+  describe('Ctrl + click', () => {
+    it('should add the clicked cells to the selection', () => {
+      render(<TestDataGridSelection />);
+      expect(document.querySelector('.Mui-selected')).to.equal(null);
+      const cell11 = getCell(1, 1);
+      fireEvent.click(cell11);
+      expect(cell11).to.have.class('Mui-selected');
+      const cell21 = getCell(2, 1);
+      fireEvent.click(cell21, { ctrlKey: true });
+      expect(cell21).to.have.class('Mui-selected');
```

Comment:
```suggestion
      expect(cell21).to.have.class('Mui-selected');
      expect(cell11).to.have.class('Mui-selected');
```

Should check if first selected cell is still selected